File: SEM5_CODE/PYTHON/BASIC/14_factorial.py
number = int(input("Enter a number: "))                     # Read the number
factorial = 1                                               # Store the factorial result

if number < 0:                                              # Check for a negative number
    print("Factorial is not defined for negative numbers")  # Display the error
else:
    for value in range(1, number + 1):                      # Visit values from 1 to number
        factorial = factorial * value                       # Multiply the current value
    print("Factorial:", factorial)                          # Display the factorial

File: SEM3_CODE/PHP/BASIC/14_factorial.php
<?php
$number = (int) readline('Enter a number: ');              // Read the number
$factorial = 1;                                            // Store the factorial result

if ($number < 0) {                                         // Check for a negative number
    echo 'Factorial is not defined for negative numbers';  // Display the error
} else {
    for ($value = 1; $value <= $number; $value++) {        // Visit values up to number
        $factorial = $factorial * $value;                  // Multiply the current value
    }
    echo 'Factorial: ' . $factorial;                       // Display the factorial
}
?>
